typescript-react/src/utils/transCardLinkData.ts

```diff
@@ -0,0 +1,71 @@
+interface Link {
+  id: number;
+  createdAt: number;
+  url: string;
+  title: string;
+  description: string;
+  imageSource: string | null;
+}
+
+interface FolderData {
+  folder: {
+    links: Array<{
+      id: number;
+      createdAt: number;
+      url: string;
+      title: string;
+      description: string;
+      imageSource: string | null;
+    }>;
+  };
+}
+
+interface SharedData {
+  data: Array<{
+    id: number;
+    created_at: number;
+    url: string;
+    title: string;
+    description: string;
+    image_source: string | null;
+  }>;
+}
+
+type InputData = FolderData | SharedData;
+
+export interface TransformedData {
+  links?: Link[];
+}
+
+export const transCardLinkData = (data: InputData): TransformedData => {
+  try {
+    if ("folder" in data) {
+      return {
+        links: data.folder.links.map(item => ({
+          id: item.id,
+          createdAt: item.createdAt,
+          url: item.url,
+          title: item.title,
+          description: item.description,
+          imageSource: item.imageSource || null,
+        })),
```

Comment:
```suggestion
        links: data.folder.links.map(({id,createdAt,url,title,description,imageSource}) => ({
          id,
          createdAt,
          url,
          title,
          description,
          imageSource: imageSource || null,
        })),
```
**구조분해할당**을 활용하면 반복되는 item이라는 프로퍼티에서 한depth 더 들어가지 않고 바로 원하는 프로퍼티에 접근할 수 있습니다.
그리고 프로퍼티의 key, value가 같다면 **단축 속성명** 이라는 문법으로도 한번 더 줄일 수 있어요

이렇게 해야해요! 가 아닌, 이런 방법도 있다~ 알고계시면 좋다~ 입니다 :-)